src/decode/lzbuffer.rs

```diff
@@ -138,13 +138,25 @@ where
     pub fn from_stream(stream: &'a mut W, dict_size: usize) -> Self {
         info!("Dict size in LZ buffer: {}", dict_size);
         Self {
-            stream,
-            buf: vec![0; dict_size],
+            stream: stream,
+            buf: Vec::new(),
             dict_size,
             cursor: 0,
             len: 0,
         }
     }
+
+    fn get(&self, index: usize) -> u8 {
+        match self.buf.get(index) {
+            Some(x) => *x,
+            None => 0,
+        }
+    }
+
+    fn get_mut(&mut self, index: usize) -> &mut u8 {
+        self.buf.resize(index + 1, 0);
```

Comment:
If `index` is less than the current size, this shrinks the buffer, which is incorrect. We probably need more test cases to make sure to capture this.